TRACING-3262 | Document OTEL TLS options (#64719)
openshift/openshift-docs: modules/distr-tracing-otel-config-collector.adoc

          receivers: [otlp]
----
<1> The OTLP gRPC endpoint. If omitted, the default `+0.0.0.0:4317+` is used.

> <2> The TLS server side configuration. Defines paths to TLS certificates. If omitted, TLS is disabled.

Reviewer: ```suggestion
<2> The server-side TLS configuration. Defines paths to TLS certificates. If omitted, TLS is disabled.
```

> <3> The path to the TLS certificate at which the server verifies a client certificate. This sets the `ClientCAs` and `ClientAuth` to `RequireAndVerifyClientCert` in the `TLSConfig`. See link:https://godoc.org/crypto/tls#Config[Golang TLS Config] for more information.

Reviewer: ```suggestion
<3> The path to the TLS certificate at which the server verifies a client certificate. This sets the `ClientCAs` and `ClientAuth` to `RequireAndVerifyClientCert` in the `TLSConfig`. See the link:https://godoc.org/crypto/tls#Config[documentation for the `Config` of the Golang TLS package] for more information.
```

Reviewer: Resolved.

> <3> Path to the TLS certificate to use by the server to verify a client certificate. (optional) This sets the ClientCAs and ClientAuth to RequireAndVerifyClientCert in the TLSConfig. Please refer to link:https://godoc.org/crypto/tls#Config[Golang TLS Config] for more information.

Reviewer: ```suggestion
<3> The path to the TLS certificate at which the server verifies a client certificate. Optional: You can set the `ClientCAs` and `ClientAuth` to `RequireAndVerifyClientCert` in the `TLSConfig`. See link:https://godoc.org/crypto/tls#Config[Golang TLS Config] for more information.
```

Author: I wanted to say that this setting is optional

> <4> Specifies the duration after which the certificate will be reloaded. If not set, it will never be reloaded. Accepts a duration string, valid time units are `ns`, `us` (or `µs`), `ms`, `s`, `m`, `h`.

Reviewer: I wouldn't recommend using `duration` unless clearly specifying the  `duration of [something]`. Perhaps replace it `time interval` like in my suggestion below?
```suggestion
<4> Specifies the time interval at which the certificate is reloaded. If not set, the certificate is never reloaded. Accepts a string with valid units of time such as `ns`, `us` (or `µs`), `ms`, `s`, `m`, `h`.
```

> <4> Specifies the time interval at which the certificate is reloaded. If not set, the certificate is never reloaded. Accepts a string containing valid units of time such as `ns`, `us` (or `µs`), `ms`, `s`, `m`, `h`.

Reviewer 2: ```suggestion
<4> Specifies the time interval at which the certificate is reloaded. If the value is not set, the certificate is never reloaded. `reload_interval` accepts a string containing valid units of time such as `ns`, `us` (or `µs`), `ms`, `s`, `m`, `h`.
```

Reviewer 2: Is this field mandatory? If yes, you can start the statement with `Required:`

Author: it's not mandatory

> <5> The OTLP HTTP endpoint. If omitted, the default `+0.0.0.0:4318+` is used.

Reviewer 2: ```suggestion
<5> The OTLP HTTP endpoint. The default value is `+0.0.0.0:4318+`.
```

> <6> The TLS server side configuration. See the `grpc` protocol configuration section for more details.

Reviewer: ```suggestion
<6> The server-side TLS configuration. See the `grpc` protocol configuration section for more details.
```

> <6> The server-side TLS configuration. See the `grpc` protocol configuration section for more details.

Reviewer 2: ```suggestion
<6> The server-side TLS configuration. For more information, see `grpc` protocol configuration section.
```

          ca_file: ca.pem
          cert_file: cert.pem
          key_file: key.pem
          insecure: false <3>
          insecure_skip_verify: false <4>
          reload_interval: 1h <5>

>           server_name_override <6>

Reviewer: ```suggestion
          server_name_override: "<string>" <6>
```

----
<1> The OTLP gRPC endpoint. If the `+https://+` scheme is used, then client transport security is enabled and overrides the `insecure` setting in the `tls`.
<2> The client side TLS configuration. Defines paths to TLS certificates.

> <3> Whether to disable client transport security. Disabled by default.

Reviewer: ```suggestion
<3> Disables client transport security when set to `true`; `false` by default.
```

> <4> Whether to skip verifying the certificate or not. Disabled by default.

Reviewer: ```suggestion
<4> Skips verifying the certificate when set to `true`; `false` by default.
```

> <5> Specifies the duration after which the certificate will be reloaded. If not set, it will never be reloaded. Accepts a duration string, valid time units are `ns`, `us` (or `µs`), `ms`, `s`, `m`, `h`.

Reviewer: ```suggestion
<5> Specifies the time interval at which the certificate is reloaded. If not set, the certificate is never reloaded. Accepts a string containing valid units of time such as `ns`, `us` (or `µs`), `ms`, `s`, `m`, `h`.
```

----
<1> The OTLP gRPC endpoint. If the `+https://+` scheme is used, then client transport security is enabled and overrides the `insecure` setting in the `tls`.
<2> The client side TLS configuration. Defines paths to TLS certificates.

> <3> Disables client transport security when set to `true`; `false` by default.

Reviewer 2: ```suggestion
<3> Disables client transport security when set to `true`. The default value is `false`.
```

> <4> Skips verifying the certificate when set to `true`; `false` by default.

Reviewer 2: ```suggestion
<4> Skips verifying the certificate when set to `true`. The default value is `false`.
```

Author: The original text was suggested by @max-cx. There should be probably agreement how to write this correctly. 

For now I will accept the suggestion.

> <5> Specifies the time interval at which the certificate is reloaded. If not set, the certificate is never reloaded. Accepts a string containing valid units of time such as `ns`, `us` (or `µs`), `ms`, `s`, `m`, `h`.

Reviewer 2: Same comments as above

> <6> When set to a non-empty string, overrides the virtual host name of authority such as the authority header field in requests. Use this for testing.

Reviewer 2: Can we also add what this field is? That would mean whatever value the user gives, the corresponding result is expected. We would have to specify `non-empty` string.
Also is `authority header` an example? If not, can we have an actual example?

```suggestion
<6> Overrides the virtual host name of authority such as the authority header field in requests. You can use this for testing.
```

> <6> If set to a non-empty string, it will override the virtual host name of authority (e.g. :authority header field) in requests (typically used for testing).

Reviewer: I'm confused about `<6>` and `<7>` now. Could you please check these two callouts in the codeblock?
```

Reviewer: ```suggestion
<6> When set to a non-empty string, overrides the virtual host name of authority such as the authority header field in requests. Use this for testing.
```

> <7> Headers are sent for every RPC performed during an established connection.

Author: remote procedure call.

I have changed it to request
